### DISCUSSION TITLE
Unable to see history of a renamed file

### DISCUSSION THREAD

**Discussion by HarelM** [community/community#3272] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi,

I have recently changed a folder in my repository to find out the entire history is now not viewable in GitHub web interface.
Please add an option to use the --follow option in the settings somewhere or enable it by default.
This issue is open for a decade now...
See here:
https://github.com/isaacs/github/issues/900
And here:
https://stackoverflow.com/questions/5646174/how-to-make-github-follow-directory-history-after-renames

This is embarrassing, honestly...

Thanks ahead,
Harel M.

**Comment by hodapp512** [comment]
This should probably be moved to a different category like general feedback rather than Mobile Feedback.

**Comment by omavi** [comment]
+1

**Comment by jacopotediosi** [comment]
+1

**Comment by dmik** [comment]
I was literally shocked when I discovered it today. I gave up on bitbucket for exactly this reason many years ago and I somehow was sure GitHub had support for this very basic VCS functionality right away. But no, which is totally strange given their scale and reputation.

For those curious GitLab supports showing history after renames perfectly well and since a very long time. For me it's a significant point to prefer GitLab over GitHub when other circumstances allow me to.

PS. As a workaround, one may use the Blame functionality that gives access to pre-rename commits in some way.

**Reply by iainelder** [reply]
+1 for the blame workaround.

**Comment by Jackenmen** [comment]
This should probably be moved to "General Feedback" category.

**Comment by ike** [comment]
Just realized that this wasn't supported, and based on the workflow that we use on the [owasp/asvs](https://github.com/OWASP/ASVS/) project, it's really important that we maintain history as we cut new versions. It's extremely inconvenient to not be able to see that history in the Github interface, and has the potential to confuse our contributors as to the nature of our release workflow.

**Comment by swharden** [comment]
+1!

Related,
* #6964

**Comment by davidstaheli** [comment, marked as answer]
Thanks to @shmeadyy and team, this feature shipped today! 🎉 More details are here: https://github.blog/changelog/2022-06-06-view-commit-history-across-file-renames-and-moves/

**Reply by mattste** [reply]
Thanks for shipping this! I got a lot of value out of this yesterday. It'd be great to have a history view that doesn't require clicking the new "Browser History" button. During a session today, I had to click the button three times to find the history I cared about.
